frontend/src/context/ClinicContext.js

```diff
@@ -0,0 +1,34 @@
+import { createContext, useReducer } from "react"
+
+export const ClinicContext = createContext()
+
+export const ClinicReducer = (state, action) => {
+    switch (action.type){
+
+    case 'SET_CLINICS':
+        return {
+                clinics: action.payload
+        }
+    case 'CREATE_CLINICS':
+        return {
+                clinics: [action.payload, ...state.clinics]
+        }
```

Comment:
I'd recommend this return to be oneliner
return { clinics: [action.payload, ...state.clinics] }